### PR TITLE
Bump Controller Runtime version to 0.6

### PR DESCRIPTION
**Pull request by aruniiird**
Signed-off-by: Arun Kumar Mohan <amohan@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by aruniiird** [comment]
@umangachapagain , I have updated the dependencies' versions to v0.19.3 . Please take a look...

**Comment by travisn** [comment]
@aruniiird Some more code updates are needed, see the errors in the CI.

**Comment by aruniiird** [comment]
Addressed all other issues, but now I have hit the following `lib-bucket-provisioner` error
```
/home/arun/go/pkg/mod/github.com/kube-object-storage/lib-bucket-provisioner@v0.0.0-20200610144127-e2eec875d6d1/pkg/client/clientset/versioned/typed/objectbucket.io/v1alpha1/objectbucket.go:71:5: not enough arguments in call to c.client.Get().Resource("objectbuckets").Name(name).VersionedParams(&options, scheme.ParameterCodec).Do
	have ()
	want (context.Context)
/home/arun/go/pkg/mod/github.com/kube-object-storage/lib-bucket-provisioner@v0.0.0-20200610144127-e2eec875d6d1/pkg/client/clientset/versioned/typed/objectbucket.io/v1alpha1/objectbucket.go:87:5: not enough arguments in call to c.client.Get().Resource("objectbuckets").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Do
	have ()
	want (context.Context)
```

**Comment by leseb** [comment]
> Addressed all other issues, but now I have hit the following `lib-bucket-provisioner` error
> 
> ```
> /home/arun/go/pkg/mod/github.com/kube-object-storage/lib-bucket-provisioner@v0.0.0-20200610144127-e2eec875d6d1/pkg/client/clientset/versioned/typed/objectbucket.io/v1alpha1/objectbucket.go:71:5: not enough arguments in call to c.client.Get().Resource("objectbuckets").Name(name).VersionedParams(&options, scheme.ParameterCodec).Do
> 	have ()
> 	want (context.Context)
> /home/arun/go/pkg/mod/github.com/kube-object-storage/lib-bucket-provisioner@v0.0.0-20200610144127-e2eec875d6d1/pkg/client/clientset/versioned/typed/objectbucket.io/v1alpha1/objectbucket.go:87:5: not enough arguments in call to c.client.Get().Resource("objectbuckets").VersionedParams(&opts, scheme.ParameterCodec).Timeout(timeout).Do
> 	have ()
> 	want (context.Context)
> ```

You might need to make a PR against https://github.com/kube-object-storage/lib-bucket-provisioner

**Comment by BlaineEXE** [comment]
Given the timing of the v1.5.0 release
1) should this block release of v1.5,
2) should this be included as an update to v1.5,
3) or should this go into v1.6?

@travisn and @leseb 
My gut check is option 3 but wanted to check in.

**Comment by leseb** [comment]
> Given the timing of the v1.5.0 release
> 
> 1. should this block release of v1.5,
> 2. should this be included as an update to v1.5,
> 3. or should this go into v1.6?
> 
> @travisn and @leseb
> My gut check is option 3 but wanted to check in.

Let's discuss today, in the meantime, I have raised https://github.com/kube-object-storage/lib-bucket-provisioner/pull/197
Once merged we should be able to get this one in.

**Comment by leseb** [comment]
@aruniiird my PR on lib-bucket-provisioner has been merged you can rebase your work on top of it.

**Comment by aruniiird** [comment]
All the `go vet` errors are taken care of. Handling `sig-storage-lib-external-provisioner` updated api errors...

**Comment by aruniiird** [comment]
All the lint / vet errors are handled...
@leseb , @travisn @BlaineEXE , please take a look

File: pkg/daemon/ceph/agent/flexvolume/attachment/crd.go [hunk 1]
```diff
@@ -46,23 +48,27 @@ func New(context *clusterd.Context) (Attachment, error) {
 
 // Get queries the Volume CRD from Kubernetes
 func (c *crd) Get(namespace, name string) (*rookalpha.Volume, error) {
-	return c.context.RookClientset.RookV1alpha2().Volumes(namespace).Get(name, metav1.GetOptions{})
+	ctx := context.TODO()
```
Comment (leseb) [inline pkg/daemon/ceph/agent/flexvolume/attachment/crd.go:51]: you don't need `ctx` just pass `context.TODO()` directly.

File: pkg/operator/ceph/operator.go [hunk 2]
```diff
@@ -120,13 +121,17 @@ func (o *Operator) Run() error {
 		}
 	}
 
+	// creating a context
+	stopContext, stopFunc := context.WithCancel(context.Background())
```
Comment (leseb) [inline pkg/operator/ceph/operator.go:125]: please rename `stopFunc` to `contextCancel` it's clearer and `stopContext` just `ctx`, it's more common.

Comment (leseb) [reply pkg/operator/ceph/operator.go:125]: also, how about using `defer` instead of calling `stopFunc()` everywhere?

**Review by leseb — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/operator.go (hunk 2 above)
Comment (aruniiird) [reply pkg/operator/ceph/operator.go:125]: Good point... Yup made the change...

File: pkg/daemon/ceph/agent/flexvolume/attachment/crd.go (hunk 1 above)
Comment (aruniiird) [reply pkg/daemon/ceph/agent/flexvolume/attachment/crd.go]: Yup done... 
Initial idea was to make changes to the `attachment.Attachment` interface itself (to match it the same with the latest api changes), but later refrained from it.

Comment (leseb) [reply pkg/daemon/ceph/agent/flexvolume/attachment/crd.go]: What's done I still see `ctx := context.TODO()`?

Comment (aruniiird) [reply pkg/daemon/ceph/agent/flexvolume/attachment/crd.go]: Corrected... I missed other lines...

Comment (leseb) [reply pkg/daemon/ceph/agent/flexvolume/attachment/crd.go]: I still don't see anything

**Comment by BlaineEXE** [comment]
How close is this to being merged? I'm building multiple fixes for https://github.com/rook/rook/issues/6650 on top of this for now.

**Review by umangachapagain — COMMENTED** [review]
Would it make sense to merge first 4 commits into 1? It's all part of single deps upgrade.

Also, let's not mention SDK here. This is more about supporting K8s 1.19 and Controller Runtime 0.6.z.

**Comment by leseb** [comment]
> How close is this to being merged? I'm building multiple fixes for #6650 on top of this for now.

It will be merged today, I want to manually test it.

**Review by leseb — APPROVED** [review]
Successfully tested the change, no side effects observed.

**Review by umangachapagain — APPROVED** [review]

**Pull request merged**